waiter/src/waiter/service_description.clj

```diff
@@ -490,6 +490,10 @@
           ; else rely on presence of x-waiter headers to set the run-as-user
           contains-waiter-header? (headers/contains-waiter-header waiter-headers on-the-fly-service-description-keys)
           user-service-description (cond-> sanitized-metadata-description
+                                           (and (contains? (:headers sources) "run-as-user")
```

Comment:
Can we move this clause below and `or` with the `(and (not (contains? sanitized-metadata-description "permitted-user")) contains-waiter-header?)` clause